docs/api/array.md

```diff
@@ -37,9 +38,18 @@ As a special case:
 - To denote a scalar shape use `""`, e.g. `Float[Array, ""]`.
 - To denote an arbitrary shape (and only check dtype) use `"..."`, e.g. `Float[Array, "..."]`.
 - You cannot have more than one use of multiple-axes, i.e. you can only use `...` or `*name` at most once in each array.
-- An example of broadcasting multiple axes:  
-    `def add(x: Float[Array, "*#foo"], y: Float[Array, "*#foo"]) -> Float[Array, "*#foo"]`.
 - A symbolic expression cannot be evaluated unless all of the axes sizes it refers to have already been processed. In practice this usually means that they should only be used in annotations for the return type, and only use axes declared in the arguments.
+- Symbolic expressions are evaluated in two stages: they are first evaluated as f-strings using the arguments of the function, and second are evaluated using the processed axis sizes. The f-string evaluation means that they can use local variables by enclosing them with curly braces, e.g. `{variable}`, e.g.
+    ```python
+    def full(size: int, fill: float) -> Float[Array, "{shape}"]:
```

Comment:
pretty sure this should be `{size}`?

Reply:
Agreed! I'll fix it now.
This does make me think, though -- handling a general `shape` is a little annoying with this change:
```python
def full(shape: tuple[int, ...], fill: float) -> Float[Array, "{strshape}"]:
    strshape = " ".join(map(str, shape))
    return jnp.full(shape, fill)
```
I don't have a good answer for that right now.

Reply:
yeah i thought about this for a minute and decided it was tough. maybe just a helper function? `jaxtyping.strshape` that just does the first line? tuple shapes are pretty common!